Mariah - Docs - Meadow Datasheets (#776)
WildernessLabs/Documentation: docs/Hardware/Reference/Meadow_Hardware/Meadow_F7/F7v2/index.md

### Battery (BAT) Pin

The battery pin (BAT) provides an alternative positive terminal connection for an external battery or power source to the built-in JST-PH battery connector. If using the BAT pin, make sure to tie the negative terminal of the battery to the board ground (GND).

### Enable (EN) Pin

The enable pin (EN) serves as a power switch for the board. By default, it is pulled HIGH (3.3V), but when pulled LOW (0V), it will disconnect most of the power to the MCU. However, it will not disconnect backup power to the MCU, so that it will keep the RTC going and keeping time. The following schematic illustrates a sample application with a switch controlling the EN pin:

![Schematic diagram of switch controlling the EN pin on the Meadow F7 Micro Dev Board, or Feather, where the switch lies between EN and GND.](ENpin.png)

### Analog Reference (AREF) Pin

The analog reference (AREF) pin provides a reference voltage for the Analog to Digital Converter (ADC) to compare against. Typically, this should be supplied with 3.3V, so as a convenience, the AREF pin is actually connected to the 3.3V rail via 0Ω resistor that is located next to the D08 pin, just below the main MCU:

> ![Photo displaying the location of the analog reference resistor.](AREFresistor.png)

Reviewer: ```suggestion
![Photo displaying the location of the analog reference resistor, left of the board's D09 and D08 pin holes, below the ARM processor.](AREFresistor.png)
```

Note that as long as the board still has power, the RTC will continue to keep time without resetting.

### Real-Time Clock (RTC)

The STM32F7 is equipped with a real-time clock (RTC), which, when set, will retain the system time as long as the board has power. If the board will have intermittent power, as when powered by a solar panel, having a battery hooked up to the board will ensure the RTC will not lose the time.

### Antenna

There is an onboard ceramic chip antenna and a U.FL connector for an external antenna for the 2.4GHz WiFi and Bluetooth radio. Additionally, there is a [SKY] antenna switch for switching between the two; by default, the chip antenna is selected, and you must use the Meadow.OS device API to switch to the external antenna.

## Peripherals and IO Pinout

Both Meadow F7v2 Feather and F7 Core-Compute share a nearly identical pinout which provides an easy upgrade path from low volume to high volume production - an application built on the F7 Feature can easily be upgraded to run on a Core-Compute module.

> ![Pinout diagram for Meadow F7v2 Feather IO.](F7v2FeatherIOPinout.png)

Reviewer: ```suggestion
![Pinout diagram for Meadow F7v2 Feather IO, as described in the following table, with RST, 3.3V, AREF, and GND at the top left. The left side ends with D04, and D05 starts at the bottom right and incrementing to D15 before ending with 5V, Enable, and Battery pins.](F7v2FeatherIOPinout.png)
```

EasyEDA symbols and footprints for a variety of use cases can be found online [here](https://easyeda.com/search?wd=meadow&indextype=components).

The following are direct links to specific schematics:

* [Meadow F7v2 Feather Schematic](https://github.com/WildernessLabs/Meadow_Hardware_Designs/blob/main/Meadow_F7v2/Feather_Dev_Module/PCA_Meadow_F7_Micro_SCH_V2b.pdf)
* [Meadow F7v2 Core-Compute Schematic](https://github.com/WildernessLabs/Meadow_Hardware_Designs/blob/main/Meadow_F7v2/Core-Compute_Module/F7CoreComputeModuleSchV2a.pdf)
* [Meadow F7v2 Core-Compute Debug Breakout Schematic](https://github.com/WildernessLabs/Meadow_Hardware_Designs/blob/main/Meadow_F7v2/Core-Compute_Module_Breakouts/Schematic_Meadow%20F7v2%20Core%20Compute%20-%20Debug%20Breakout.pdf)

### Dimensions and Weight - F7v2 Feather

Basic dimensions are as follows; for 3D CAD models and extended dimensional information, please see the [Wilderness Labs 3D Designs Git Repository](https://github.com/WildernessLabs/3D_Print_Designs/tree/master/Meadow/Reference_Models).

> ![Dimensions model for F7v2 Feather.](featherDimensions.png)

Reviewer: ```suggestion
![Layout diagram model for F7v2 Feather showing a width of 22.86 millimeters (0.9 inches) and length of 73.66 millimeters (2.9 inches). Pin holes are 0.04 inches diameter (1.02 millimeters) and are 0.1 inches (2.54 millimeters) from center to center. And corner mounting holes are 0.1 inches (2.54 millimeters) from the sides.](featherDimensions.png)
```

* Width = 0.9” [22.86mm]
* Length = 2.9” [73.66mm]
* Weight = 0.3oz [7.0 grams]

### Schematic Symbol - F7v2 Feather

> ![Schematic Symbol for Meadow F7v2 Feather.](featherSchematicSymbol.png)

Reviewer: ```suggestion
![Schematic symbol diagram for Meadow F7v2 Feather with RST, 3.3V, AREF, and GND at the top left. The left side ends with ESP32_RST on pin 38. The bottom right starts with ESP32_MTMS at pin 39 and goes to the top right with D15 at pin 29 before ending with pins for 5V, Enable, and Battery.](featherSchematicSymbol.png)
```

### Dimensions and Weight - F7v2 Core-Compute Module

> ![Dimensions model for F7v2 Core-Compute Module.](ccmDimensions.png)

Reviewer: ```suggestion
![Dimensions diagram for F7v2 Core-Compute Module. The board is 35.5 millimeters long and 24.5 millimeters wide.](ccmDimensions.png)
```

* Width = 24.5mm
* Length = 35.5mm
* Height = 1.6mm to top of PCB, 3mm total height clearance required
* Weight = 0.1oz [3.0 grams]

### Schematic Symbols and Footprints - F7v2 Core-Compute Module

#### Physical Layout

The following schematic symbol represents the physcial pin layout:

> ![Schematic symbol of Meadow Core-Compute Module physical pin layout.](ccmPhysicalLayout.png)

Reviewer: ```suggestion
![Schematic symbol of Meadow Core-Compute Module physical pin layout, starting bottom left with GND at pin 90 up to GND at pin 57, the top with pin 56 (D15) over to OTG_VBUS at pin 35, then down the right starting at ETH_IRQ pin 34 to GND at pin 1.](ccmPhysicalLayout.png)
```

#### Logical Layout

The following schematic symbol provides a suggested logical layout of pins based on primary function:

> ![Schematic symbol of Meadow Core-Compute Module using a suggested logical layout of pins.](ccmLogicalLayout.png)

Reviewer: ```suggestion
![Schematic symbol of Meadow Core-Compute Module using a suggested logical layout of pins, with sections for analog, SPI3, COM4, CAN1, ethernet, and SD card on the left, and debug, I2C1 COM1, SPI5, I2C3, and OTG USB on the right.](ccmLogicalLayout.png)
```

#### Recommended PCB Footprint

> ![Symbol of recommended PCB footprint for Meadow Core-Compute Module.](ccmPCBfootprint.png)

Reviewer: ```suggestion
![Diagram of recommended PCB footprint for Meadow Core-Compute Module, with an outside pad of 1.2 by 0.5 millimeters and PAD91 of 1.5 by 1.0 millimeters.](ccmPCBfootprint.png)
```
